### DISCUSSION TITLE
Publishing a Java library using JReleaser - maven.nexus2.mavenCentral.stagingRepository does not exist

### DISCUSSION THREAD

**Discussion by lsoares** [jreleaser/jreleaser#1598] [Q&A] [answered]
what is the `stagingRepository`? 
I was having the following issue:
https://stackoverflow.com/questions/77997426/publishing-a-java-library-using-jreleaser-maven-nexus2-mavencentral-stagingrep

When I set `stagingRepository` to `"build/jreleaser"` (just trial and error) I get a different error:

```
[ERROR] JReleaser failed after 0.015 s

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':lib:jreleaserFullRelease'.
> Cannot invoke "java.nio.file.Path.getFileName()" because "p" is null
```

can anyone help?

**Comment by aalmiray** [comment, marked as answer]
The `stagingRepository` is a local directory that contains artifacts to be deployed. Its use is explained at the guide https://jreleaser.org/guide/latest/examples/maven/maven-central.html#_gradle
